src/applications/static-pages/health-care-manage-benefits/components/CernerCallToAction/index.js

```diff
@@ -0,0 +1,165 @@
+  }
+
+  fetchFacilities = async facilityIDs => {
+    // Show loading state.
+    this.setState({ fetching: true });
+
+    try {
+      // Fetch facilities and store them in state.
+      const response = await apiRequest(
+        `${environment.API_URL}/v1/facilities/va?ids=${facilityIDs.join(',')}`,
+      );
+      this.setState({ facilities: response?.data, fetching: false });
+
+      // Log any API errors.
+    } catch (error) {
+      this.setState({ error: error.message, fetching: false });
+    }
+  };
+
+  render() {
+    const { callToActions, text, type } = this.props;
+    const { error, fetching, facilities } = this.state;
+
+    // Escape early if we are fetching.
+    if (fetching) {
+      return <LoadingIndicator message="Loading your information..." />;
+    }
+
+    // Escape early if there was an error fetching the Cerner facilities.
+    if (error || isEmpty(facilities)) {
+      // WARNING: Add sentry logging here if there is an error fetching Cerner facilities.
+      return (
+        <AlertBox
+          headline="Something went wrong"
+          content="We’re sorry. Something went wrong on our end. Please try again later."
+          status="error"
+        />
+      );
+    }
+
+    // Derive the Cerner facility names.
+    const facilityNames = map(
+      facilities,
+      facility => facility?.attributes?.name || 'unknown facility name',
+    );
+    const joinedFacilityNames =
+      facilityNames.join(', ') || 'Cerner facility(s)';
+
+    return (
+      <div className="usa-alert usa-alert-warning">
+        <div className="usa-alert-body">
+          <h3 className="usa-alert-heading">
+            According to our records, you are registered at a clinic within{' '}
+            {joinedFacilityNames}. VA providers at this facility and its clinics
+            are using the new My VA Health portal.
+          </h3>
+          <p className="usa-alert-text vads-u-margin-y--4">
+            You may need to sign in again to view your VA lab and test results.
+            If you do, please sign in with the same account you used to sign in
+            here on VA.gov. You also may need to disable your browser&apos;s
+            pop-up blocker so that {type} tools are able to open.
+          </p>
+          {map(callToActions, callToAction => {
+            // Derive callToAction properties.
+            const headerText = callToAction?.deriveHeaderText(
+              joinedFacilityNames,
+            );
+            const href = callToAction?.href;
+            const label = callToAction?.label;
+
+            return (
+              <>
+                {headerText && (
+                  <h3 className="usa-alert-heading">{headerText}</h3>
+                )}
+                <a
+                  className="usa-button vads-u-color--white"
+                  href={href}
+                  rel="noopener noreferrer"
+                >
+                  {label}
+                </a>
```

Comment:
It seems like this component is more complex because we have to load up the user facilities and want to share content between the different widgets, but I'm wonder what your thoughts are on restructuring to a component that only loads up the joined facilities, then a `render` prop to just let the parent component decide what to do with that. This could make the rendering logic more straightforward and the content itself more flexible, in case one of them changes.

```jsx
<LoadUserFacilities
     render={(joinedFacilityNames) => {
         return (
              // DOM without dynamic behavior
         );
     })
/>
```

Reply:
Oh! Great idea! Can we keep this in mind for a future iteration as I'll have to update quite a bit of code to achieve this? I really do like the idea! 🔥